### PR TITLE
DefaultArgParser.py: Improve filter-by usage

### PR DESCRIPTION
**Pull request by troublemagnet**
Changes the usage string of filter-by for better understanding.

Closes https://github.com/coala/coala/issues/4597

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/parsing/DefaultArgParser.py [hunk 1]
```diff
@@ -189,6 +189,7 @@ def default_arg_parser(formatter_class=None):
 
     outputs_group.add_argument(
         '--filter-by', action='append', nargs='+',
+        metavar=('FILTER_NAME', 'FILTER_ARGS'),
```
Comment (shreyans800755) [inline coalib/parsing/DefaultArgParser.py]: At least one filter arg has to be present.
So, output should be changed like this:
`[--filter-by FILTER_NAME [FILTER_ARGS ...]` -> `[--filter-by FILTER_NAME FILTER_ARG [FILTER_ARG ...]`

**Comment by shreyans800755** [comment]
@troublemagnet 
Please update the description of the commit:
`Changes the usage string of filter-by for better understanding.` -> `Improves the usage string of filter-by option.`

**Comment by Makman2** [comment]
Change looks good, however please squash the commits together and rebase ;)

**Comment by shreyans800755** [comment]
ack ad65269

**Comment by shreyans800755** [comment]
@Makman2 
I've tried this changes in my local and it looks good. IMO it can be merged.

**Comment by Makman2** [comment]
ack 34c50d6

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**